independent-projects/tools/analytics-common/src/main/java/io/quarkus/analytics/util/FileUtils.java

```diff
@@ -78,7 +78,7 @@ public static <T> Optional<T> read(Class<T> clazz, Path path, MessageWriter log)
         try {
             final ObjectMapper mapper = getObjectMapper();
             return Optional.of(mapper.readValue(path.toFile(), clazz));
-        } catch (Exception e) {
+        } catch (Throwable e) {
             log.warn("[Quarkus build analytics] Could not read {}", path.toString(), e);
```

Comment:
the log/warn for reading the file is true for Exception.

For Throwable - that's a much bigger net. Should we maybe instead add separate catch of throwable and make it log an more explicit error ? like: `log.error("[Quarkus build analytic] Unexpected error " + e.getClass().getName() + ": "  + e.getMessage() + ". Attempting to continue"); or similar?

in case its an outofmemory or nosuchfieldexpection or similar that is completely unrelated to what could be expected error (like file not found or corrupt)

Reply:
I'm fine with that

Reply:
done